hngprojects/hng_boilerplate_nestjs
# Feat: Export User Data Endpoint

**What does the PR do?**

This PR introduces a new API endpoint `/api/v1/export/user` for exporting user data. It includes server-side logic to fetch user data along with their associated organisations and anything related to user and convert the data into a JSON format for export.

**How should this be manually tested?**

Send a GET request to `/api/v1/export/user?userId=USER_ID` where `USER_ID` is the ID of the user you want to export. Ensure that the response contains the user data and associated organisations in JSON format.

**Checklist of what you did**

- [x] Developed server-side logic for the `/api/v1/export/user` endpoint.
- [x] Integrated user data fetching with relations to `owned_organisations` and `created_organisations`.
- [x] Implemented JSON export functionality.
- [x] Added Swagger documentation for the new endpoint.
- [x] Wrote unit tests to verify the functionality of the export service.
- [x] Tested various scenarios to ensure data is correctly exported and errors are handled.

**Reference Issue**

[FEAT] Export User Data Endpoint #138  (https://github.com/hngprojects/hng_boilerplate_nestjs/issues/138)
